### PR TITLE
1201 attention unet

### LINKED ISSUE #1201: ATTENTION BASED UNET
It would be very good if the project could provide a UNET model with attention mechanism like the one described in https://github.com/ozan-oktay/Attention-Gated-Networks.
I think it would improve the performance of segmentation models

### PR DESCRIPTION
**Pull request by juampatronics**
Signed-off-by: Juan Pablo de la Cruz Gutiérrez <juampatronics@gmail.com>

Fixes #1201

### Description
Implementation of attention based Unet following the description of the referred paper.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ x] In-line docstrings updated.
- [ x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by wyli** [comment]
thanks, the PR looks nice in general, just two issues from the CI:
- `padding: str` in `Conv*d` is not supported for pytorch<1.9: please see the error msg here https://github.com/Project-MONAI/MONAI/runs/5561083759?check_suite_focus=true, the network should either raise a runtime exception for those pytorch versions, or be changed to have a compatible parameter if possible
- there are some style issues: https://github.com/Project-MONAI/MONAI/runs/5561083011?check_suite_focus=true

please let us know if any questions for these.

File: monai/networks/nets/attentionunet.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+                dropout=self.dropout,
+            ),
+            dropout=self.dropout,
+        )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return self.model(x)
+
+
+class AttentionLayer(nn.Module):
```
Comment (ericspod) [inline monai/networks/nets/attentionunet.py:242]: It's a little easier to read if this is above the network definition where it's used.

File: monai/networks/nets/attentionunet.py [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+        psi = self.relu(g1 + x1)
+        psi = self.psi(psi)
+
+        return x * psi
+
+
+@export("monai.networks.nets")
+@alias("Attentionunet")
+@alias("AttentionUNet")
+class AttentionUnet(nn.Module):
```
Comment (ericspod) [inline monai/networks/nets/attentionunet.py:147]: I think we could tweak this class to inherit from our UNet class but we'd want to change [this line](https://github.com/Project-MONAI/MONAI/blob/33f632a007469a9581c1c5f42a4823d6b1e1dca0/monai/networks/nets/unet.py#L189) on UNet to be another template method which you could then override. I'm good with merging this version now and we can consider merging UNet definitions of various networks we have later.

File: monai/networks/nets/attentionunet.py (hunk 1 above)
Comment (juampatronics) [reply monai/networks/nets/attentionunet.py]: Done

**Review by wyli — APPROVED** [review]
